python/cuml/dask/ensemble/randomforestregressor.py

```diff
@@ -0,0 +1,327 @@
+                             split_algo, split_criterion,
+                             bootstrap, bootstrap_features,
+                             verbose, min_rows_per_node,
+                             rows_sample, max_leaves,
+                             accuracy_metric, quantile_per_tree,
+                             random.random(),
+                             workers=[worker])
+            for worker, n in ws}
+        
+        rfs_wait = list()
+        for r in self.rfs.values():
+            rfs_wait.append(r)
+                
+        wait(rfs_wait)
+        
+        
+    @staticmethod
+    def _func_build_rf(n, n_estimators, max_depth,
+                 handle, max_features, n_bins, 
+                 split_algo, split_criterion,
+                 bootstrap, bootstrap_features,
+                 verbose, min_rows_per_node,
+                 rows_sample, max_leaves,
+                 accuracy_metric, quantile_per_tree,
+                 r):
+        
+        return cuRFR(n_estimators=n_estimators, max_depth=max_depth, handle=handle,
+                 max_features=max_features, n_bins=n_bins, 
+                 split_algo=split_algo, split_criterion=split_criterion,
+                 bootstrap=bootstrap, bootstrap_features=bootstrap_features,
+                 verbose=verbose, min_rows_per_node=min_rows_per_node,
+                 rows_sample=rows_sample, max_leaves=max_leaves,
+                 accuracy_metric=accuracy_metric, 
+                 quantile_per_tree=quantile_per_tree)
+                
+    
+    @staticmethod
+    def _fit(model, X_df, y_df, r): 
+        return model.fit(X_df, y_df)
+    
+    @staticmethod
+    def _predict(model, X, r): 
+        return model.predict(X)
+    
+    def fit(self, X, y):
+        """
+        Fit the input data to Random Forest classifier
+
+        Parameters
+        ----------
+        X : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, n_features).
+        y : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, 1)
+        """
+        c = default_client()
+
+        X_futures = c.sync(_extract_ddf_partitions, X)
+        y_futures = c.sync(_extract_ddf_partitions, y)
+                                       
+        f = list()
+        for w, xc in X_futures.items():     
+            f.append(c.submit(RandomForestRegressor._fit, self.rfs[w], xc, y_futures[w], random.random(),
+                             workers=[w]))            
+                               
+        wait(f)
+        
+        return self
+    
+    def predict(self, X):              
+        """
+        Predicts the labels for X.
+
+        Parameters
+        ----------
+        X : Acceptable format: dask-cudf. Dense matrix (floats or doubles) of shape (n_samples, n_features).
+
+        Returns
+        ----------
+        y: NumPy
+           Dense vector (int) of shape (n_samples, 1)
```

Comment:
General question, would we not want an option of returning a device distributed data format (dask cudf for example)?

Reply:
I am not sure about this right now because the inference is performed on CPU. Maybe, we can distribute results and create a dask cudf.

Reply:
We definitely want to follow the GPU in GPU out paradigm here. I believe the RF algorithm should be returning predictions on GPU even though they are being computed on host. Is this not the case?